src/main/kotlin/com/manimdsl/runtime/ASTExecutor.kt

```diff
@@ -236,15 +249,32 @@ class VirtualMachine(
                 RuntimeError(value = "Array index out of bounds", lineNumber = arrayElemNode.lineNumber)
             } else {
                 arrayValue.array[index.value.toInt()] = assignedValue
-                linearRepresentation.add(ArrayElemRestyle((arrayValue.manimObject as ArrayStructure).ident, listOf(index.value.toInt()), arrayValue.style.animate!!))
+                arrayValue.animatedStyle?.let {
+                    linearRepresentation.add(
+                        ArrayElemRestyle(
+                            (arrayValue.manimObject as ArrayStructure).ident,
+                            listOf(index.value.toInt()),
+                            it
+                        )
+                    )
+                }
                 linearRepresentation.add(
                     ArrayElemAssignObject(
                         (arrayValue.manimObject as ArrayStructure).ident,
                         index.value.toInt(),
-                        assignedValue
+                        assignedValue,
+                        arrayValue.animatedStyle
                     )
                 )
-                linearRepresentation.add(ArrayElemRestyle((arrayValue.manimObject as ArrayStructure).ident, listOf(index.value.toInt()), arrayValue.style))
+                if (arrayValue.animatedStyle != null) {
```

Comment:
It should only be styling 'back to normal' if there was an animated style to start with, which is why I added this check